graphbolt/src/random.cc

```diff
@@ -59,6 +61,10 @@ void RandomEngine::SetSeed(uint64_t seed, uint64_t stream) {
 }
 
 /** @brief Manually fix the seed. */
-void RandomEngine::SetManualSeed(int64_t seed) { manual_seed = seed; }
+void RandomEngine::SetManualSeed(int64_t seed) {
+  RandomEngine::ThreadLocal()->SetSeed(seed);
```

Comment:
could you add a comment here that we set seed for current thread intentionally. this is the root cause of the issue?

Reply:
Yes, that was the root cause. Let me do that. The mutex is just to make the code future-proof.